### PR TITLE
Validate restoring a 4.3.x backup to a GPDB 5 database

### PR DESCRIPTION
**Pull request by jmcatamney**
This series of commits splits up the single backup.feature file (where each scenario both dumps and restores data) into two files, dumps_only.feature which performs all necessary dumps and is run against a 4.3 database and restores_only.feature which performs the corresponding restores and is run against a 5 database.  It also adds and/or modifies Behave steps to account for differences in functionality between 4.3 and 5.

Authors: Chris Hajas, Karen Huddleston, Jamie McAtamney

### CODE REVIEW

File: gpMgmt/test/behave_utils/utils.py [hunk 1]
```diff
@@ -1380,8 +1401,14 @@ def verify_restored_table_is_analyzed(context, table_name, dbname):
     else:
         schema_name = 'public'
     with dbconn.connect(dbconn.DbURL(dbname=dbname)) as conn:
-        schema_name = escape_string(schema_name, conn=conn)
-        table_name = escape_string(table_name, conn=conn) 
+        if have_escape_string:
+            schema_name = escape_string(schema_name, conn=conn)
+            table_name = escape_string(table_name, conn=conn)
+        else:
+            return False
```
Comment (khuddlefish) [inline gpMgmt/test/behave_utils/utils.py]: Why is this return false here?

File: gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py [hunk 2]
```diff
@@ -299,7 +383,6 @@ def impl(context, HOST, PORT, USER, transition):
         run_command(context, trigger_transition)
         wait_till_change_tracking_transition(host,port,user)
     if transition == 'resync':
-        command = 'gpfaultinjector -f filerep_consumer -m async -y fault -r primary -H ALL'
```
Comment (professor) [inline gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py]: Why is this removed?
